packages/frontend/src/components/create-campaign/form/rewards-step/reward-tokens-list/index.tsx

```diff
@@ -0,0 +1,173 @@
+import { useEffect, useMemo, useRef } from "react";
+import {
+    useAccount,
+    useBlockNumber,
+    useChainId,
+    useReadContracts,
+} from "wagmi";
+import { erc20Abi } from "viem";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList } from "react-window";
+import { useTranslations } from "next-intl";
+import {
+    type Token,
+    type TokenAmount,
+    type WhitelistedErc20Token,
+    type WhitelistedErc20TokenWithBalance,
+} from "@metrom-xyz/sdk";
+import { Typography } from "@/src/ui/typography";
+import { useWhitelistedRewardsTokens } from "@/src/hooks/useWhitelistedRewardsTokens";
+import { Row } from "./row";
+
+import styles from "./styles.module.css";
+
+interface RewardTokensListProps {
+    value?: Token;
+    unavailable?: TokenAmount[];
+    onRewardTokenClick: (token: WhitelistedErc20Token) => void;
+}
+
+export function RewardTokensList({
+    value,
+    unavailable,
+    onRewardTokenClick,
+}: RewardTokensListProps) {
+    const t = useTranslations("newCampaign.form.rewards");
+    const listRef = useRef(null);
+
+    const { address } = useAccount();
+    const chainId = useChainId();
+    const { whitelistedTokens, loading } = useWhitelistedRewardsTokens();
+
+    const { data: blockNumber } = useBlockNumber({ watch: true });
+    // TODO: check if this works well with a lot of whitelisted tokens
```

Comment:
About this, pretty sure we're fine. It's highly improbable we will have more than let's say 15 whitelisted tokens at once.